CDCgov/prime-reportstream
# 16148: update FHIR reciever logic to be simpler

This PR simplifies the FHIR receiver and removes the few bugs in sender parsing and event tracking.  The simplification can be done because the FHIR Converter handles the same cases more robustly

Test Steps:
1. Turn on the submissions service and reportstream
2. Send a report to submissions
3. Verify it gets picked up by report stream

## Changes
- Drops event handling for errors in favor of putting messages on to the queue directly
- Removes the functionality duplicated with FHIR converter

## Checklist

### Testing

### Process

## Linked Issues
- #16148 

## To Be Done
TBD: but there might be another follow up ticket to address this more efficiently and remove the FHIR Receiver step, but it would require some updates to the how the item lineage works

## Specific Security-related subjects a reviewer should pay specific attention to
